sklearn/utils/fixes.py

```diff
@@ -307,3 +307,14 @@ def _contents(data_module):
     from scipy.integrate import trapezoid  # type: ignore  # noqa
 except ImportError:
     from scipy.integrate import trapz as trapezoid  # type: ignore  # noqa
+
+
+# TODO: Remove when Pandas > 2.2 is the minimum supported version
+def pd_fillna(pd, frame):
+    pd_version = parse_version(pd.__version__).base_version
+    if parse_version(pd_version) < parse_version("2.2"):
+        frame = frame.fillna(value=np.nan)
+    else:
+        with pd.option_context("future.no_silent_downcasting", True):
+            frame = frame.fillna(value=np.nan).infer_objects(copy=False)
```

Comment:
shouldn't I be expecting this to be covered?

Reply:
Quickly looking at it, it seems that if the tests fail the coverage is not uploaded ...

Reply:
It is covered only with `scipy-dev` currently.